### DISCUSSION TITLE
[QA] Why does Zitadel query the instance_id 4k times an hour

### DISCUSSION THREAD

**Discussion by SchillingSebastian** [zitadel/zitadel#6304] [Q&A] [answered]
Hey there, 

I am using a self hosted instance of Zitadel in conjunction with CockroachDB. As CockroachDB Cloud counts the RUs I am was wondering why a running instance with basically no login happening was causing so many SELECTs. 
After digging a bit I did find out, that there is a really high number of such SQL queries
```SQL
SELECT DISTINCT instance_id
 FROM eventstore.events AS OF SYSTEM TIME '_'
   WHERE ((instance_id != ALL ($1))
    AND (creation_date > $1))
```
happening. To be a bit more precise: About 4k of those statements in an hour with next to none logins or other activities.
What is this query used for and might this be a result of a faulty Zitadel configuration on my side? Can I somehow reduce this number?

Here is a screenshot of the CockroachDB SQL activity monitor:
<img width="1389" alt="image" src="https://github.com/zitadel/zitadel/assets/25250527/3bdff566-4d91-4fb7-9e1e-0a47dede1b9f">

PS: 
I did upgrade to the latest ([v2.31.4](https://github.com/zitadel/zitadel/releases/tag/v2.31.4)) version in hopes of https://github.com/zitadel/zitadel/pull/6280 reducing this number, but it still has this many executions.

**Comment by fforootd** [comment, marked as answer]
We have background jobs called "spoolers" who regularly check if everything is up to date in the normalised views of the databases.

I think this value here has the biggest impact right now https://github.com/zitadel/zitadel/blob/4d09409328f83c9b1d312ee3af261cca8fe0a64f/cmd/defaults.yaml#L176

Rest assured that we are working on a solution to reduce these calls in the near future 😆 as we speak we optimized the write side of zitadel and as soon as this is released we will improve the read side as well.
